### DISCUSSION TITLE
Instructor manual

### DISCUSSION THREAD

**Discussion by profvjreddi** [harvard-edge/cs249r_book#466] [Chapter Discussions]
Some people have asked for an instructor manual to help them use the resource in classes. It would be nice to have that in the book.

**Comment by profvjreddi** [comment]
Initial thoughts on what to put in there based on some feedback, welcome thoughts and suggestions.

1. Learning objectives for each chapter
2. Chapter-by-chapter teaching guidelines
3. Suggested lecture outlines
4. Practical exercises or projects related to each chapter
5. Solutions to end-of-chapter problems (if applicable)
6. Discussion questions and answers
7. Slides or presentation materials for each chapter
8. Suggestions for course structure
9. Updates and errata

**Reply by mioluigi** [reply]
From theory to colab (practice) at the end of each chapter.

**Reply by profvjreddi** [reply]
I agree @mioluigi, I am working on something and plan to release it by the end of the semester!

**Comment by profvjreddi** [comment]
@Mjrovai -- "We can explicitly ask them to incentivize students to put their hands on the Labs."

**Comment by mioluigi** [comment]
Wonderful news!

I thank you so much.  Please let me know how I can assist you and provide
value.  I am planning to incorporate your book into all of my classes.


Best wishes,
Luis Varela
AI, Cybersecurity, and Data Science Teacher/Consultant

**Reply by profvjreddi** [reply]
Please don't try it yet -- but this is a sneak peek of what's to come -- https://mlsysbook.org/TinyTorch/intro.html -- they basically build a full TinyTorch so they learn the engineering of these systems, and it would correspond to the parts of the book chapters. 

At a high level welcome any quick thoughts and feedback but again it is very much **work under construction.** The bugs are giving me a headache 😓

**Reply by BA-CalderonMorales** [reply]
At first glance, the ui would do a lot better without a ton of emojis. This is a preference for me and many professionals out there. Not sure what you think about this. I still need to dif into content, just noticed you were looking for feedback. 😅

**Reply by profvjreddi** [reply]
Haha, yeah, I have to say I agree with that feedback :) 

It's a work in progress, and for grunt work like that, I toss it over to vibe coding, so it ends up with that fluff. 

I will clean it up once I get back to the core material.
